invenio_stats/tasks.py

```diff
@@ -38,3 +43,142 @@ def process_events(event_types):
+
+    def get_bookmark(self):
+        """Get last aggregation date."""
+        if not Index(self.aggregation_alias,
+                     using=self.client).exists():
+            oldest_index = self.get_indices_with_alias('events-stats_{}'.
+                                                       format(self.event))
+            return datetime.datetime.strptime(oldest_index,
+                                              'events-stats_{}-%Y.%m.%d'.
+                                              format(self.event)).date()
+
+        import ipdb
+        ipdb.set_trace()
+        query_bookmark = Search(using=self.client,
+                                index=self.aggregation_alias,
+                                doc_type='{0}_bookmark'.format(self.event))
+        query_bookmark.aggs.metric('latest', 'max', field='date')
+        bookmark = query_bookmark.execute()[0]
+        bookmark = datetime.datetime.strptime(bookmark.date, '%Y-%m-%d')
+        return bookmark.date()
+
+    def set_bookmark(self):
+        """Set bookmark for starting next aggregation."""
+        def _success_date():
+            bookmark = {
+                'date': datetime.date.today().isoformat()
+            }
+            yield dict(_index='{0}-{1}-{2}'.
+                       format(self.time_frame,
+                              self.event,
+                              datetime.date.today().strftime('%Y-%m')),
+                       _type='{0}_bookmark'.format(self.event),
+                       _source=bookmark)
+
+        bulk(self.client,
+             _success_date(),
+             stats_only=True)
+
+    def pending_dates(self):
+        """Get all dates that haven't been processed."""
+        start_date = self.get_bookmark()
+        end_date = datetime.date.today()
+        import ipdb
+        ipdb.set_trace()
+        if start_date >= end_date:
+            for n in range((start_date - end_date).days + 1):
+                yield start_date + datetime.timedelta(n)
+        else:
+            for n in range((end_date - start_date).days + 1):
+                yield start_date - datetime.timedelta(n)
+
+    def aggr_iter(self, date):
+        """Aggregate and return dictionary to be indexed in ES."""
+        aggregation_data = {}
+        result = self.query.execute()
+
+        import ipdb
+        ipdb.set_trace()
+
+        for aggregation in result.aggregations:
+            for hit in result.aggregations[aggregation]['buckets']:
+                aggregation_data['bucket'] = hit['key']
+                aggregation_data['count'] = hit['doc_count']
+                print('aggregation_data:', aggregation_data)
+
+                yield dict(_id='{0}_{1}'.
+                           format(hit['key'],
+                                  date),
+                           _index='{0}-{1}-{2}'.
+                           format(self.time_frame,
+                                  self.event,
+                                  date.strftime('%Y-%m')),
```

Comment:
No need to use `time_frame`, instead we can just let the user configure the `'%Y-%m'` part.